[IV-25-26] Objetivo 2

oscar0310 opened the pull request:
## Checklist Desarrollador
Sobre la estructura del repositorio

* [x] ¿He seguido las mejores prácticas en el nombre de las clases y ficheros y
   disposición de los mismos?
* [x] ¿He mirado tanto reglas generales (para cualquier repo) como reglas
específicas para el lenguaje con el que se está trabajando?

Sobre el planteamiento

* [x] ¿Se han planteado una serie de issues, asignados a las historias de
usuario pertinentes, que clarifiquen qué es lo que necesitan los usuarios e
identifiquen las diferentes partes del problema?
* [x] ¿Los issues representan un problema, y no una tarea?

Sobre el análisis del problema

* [x] ¿Se ha documentado qué análisis se ha hecho sobre el dominio para decir lo
   que se ha creado?
* [x] ¿Se ha documentado por qué se ha elegido que lo creado sea un objeto valor,
   una entidad o un agregado?
* [x] ¿He propuesto un producto mínimamente viable, que en muchos casos será un solo
   objeto valor que no dependa de ningún otro (y que sea la base de muchos
   otros)?.

Sobre el código

* [x] ¿Todos los mensajes de commit explican el cambio, y no se
      limitan a repetir el nombre del fichero que se ha cambiado?
* [x] ¿Los mensajes de commit siguen el formato estándar y buenas prácticas?
* [x] ¿Se ha hecho una revisión real del código para comprobar que todos
   los atributos y funciones creadas están respaldadas por una HU?
* [x] ¿Todos los issues creados están relacionados con a una HU?
* [x] ¿Ha asignado el propietario/la propietaria del repositorio todos los
issues planteados al *milestone* correspondiente?
* [x] ¿Todos los cambios en el código están asignados a un issue al que se
      referencia en un commit?
* [x] ¿Se ha asignado al mismo milestone el PR que se ha hecho?
* [x] ¿Es el milestone sobre el que estoy trabajando un producto
   mínimamente viable? ¿O tengo que solicitar al product manager que precise de
   qué producto se trata?
* [x] ¿Se ha comprobado que el código entregado sea sintácticamente
correcto?

## Checklist Propietario:

* [x] ¿Los issues que ha planteado la persona son efectivamente problemas y
evidencian comprensión de las historias de usuario? Si no es así, es mejor que
se le transmita antes de que empiece a crear cualquier código erróneo.
* [x] ¿Has comprobado que efectivamente el compañero/a ha seguido *su* lista de
      comprobación?
* [x] ¿El código creado es una modelización válida de las historias de usuario
      creadas por ti, y te sirve como base para programar la lógica de negocio?
* [x] ¿El código creado sigue las mejores prácticas para el lenguaje de
      programación que hayáis consensuado, tanto en identificadores, como
      nombres de ficheros, como disposición de los diferentes ficheros en el
      repositorio? Por ejemplo, ¿todos los commits se refieren a un issue y
      aportan código para resolver el problema planteado en el mismo?
* [x] ¿He comprobado que el código compile, o que quien lo ha programado ha lo
ha hecho y documentado en el repositorio?
* [x] ¿He añadido el PR y todos los issues que se resuelvan en él al milestone
correspondiente, que será el primero?
* [x] ¿Cubre totalmente el milestone 0, o hará falta modelización adicional en
el objetivo 4 para poder programar la lógica de negocio?
* [x] En resumen, ¿considero que con lo hecho puedo ponerme a trabajar
directamente en el siguiente milestone, que incluye la lógica de negocio?

Este PR closes #6

oscar0310 committed:
fix #6 Añadiendo lenguaje

Se añade el lemguaje que se va a usar en el proyecto

oscar0310 committed:
fix #8 añadiendo directorio lib

oscar0310 committed:
Fix #9 Añadiendo el archivo CVE.pm

Representará los CVE

oscar0310 committed:
Fix #9 Cambiando nombre  CVE.pm a cve.pm

Siguiendo las buenas prácticas.

oscar0310 committed:
Fix #9 Añadiendo cve.pm

JJ commented:
Revisión: @gosema

chelunike commented:
@oscar0310  Creo que puedes describir un poco más los datos que se necesitan en el issue #9 Y luego puedes hacer commit con las clases.

oscar0310 committed:
fix #9 añadienco la clase cve

oscar0310 committed:
fix #8 gitignore de acuerdo con Perl

Se ha actualizado el gitignore para Perl

oscar0310 committed:
fix #9 eliminando cve.pm closes #9

oscar0310 committed:
Cerrando issue no valido Closes #9

oscar0310 committed:
fix #11 añadiendo entidad ctf

oscar0310 committed:
fix #11 creando ctf.pm

oscar0310 committed:
fix #11 añadiendo tipo a ctf

contiene el tipo de reto CTF

oscar0310 committed:
fix #11 añadiendo ip  a ctf

contiene la dirección ip del sistema de el reto CTF

oscar0310 committed:
fix #11 añadiendo port a ctf

contiene el puerto  del sistema de el reto CTF

oscar0310 committed:
fix #12 añadiendo la entidad cve

oscar0310 committed:
fix #12 añadiendo description a CVE

Contendrá la descripcion del CVE

JJ commented:
El problema con dedicarle tiempo a modelizar un reto es que aparte de que puede que no sea esa la forma como se va a emplear, sino simplemente argumentos separados a una función, estás gastando tiempo en cosas que no son esenciales a la lógica de negocio, sino simples argumentos que se van a emplear en la misma.

oscar0310 committed:
fix #11 eliminando ctf.pm

oscar0310 committed:
fix #11 eliminando la entidad ctf

oscar0310 committed:
fix #12 añadiendo links a la entidad cve

Contiene los enlaces que contendrán documentación que nos ayudará a implementar una forma de explotar dicha vulnerabilidad.

oscar0310 committed:
fix #12 eliminando cve.pm

oscar0310 committed:
fix #13 añadiendo la entidad cve a iv.yalm

oscar0310 committed:
#13 creando la entidad cve

oscar0310 committed:
#13 creando el atributo description de cve

Se almacenará la descripción del cve en un string que no puede ser vacío y es requerido

oscar0310 committed:
#13 añadiendo el atributo links a cve

Almacenará los links relacionados con el cve en un array de string no vacíos, es requerido

oscar0310 committed:
#13 añadiendo ; a has links

El no ponerlo provocará erroes

oscar0310 commented:
buenas @chelunike, podrías echarle un vistazo y darme feedback.

oscar0310 commented:
@chelunike me haría falta que miraras lo que he hecho por si habría que hacerle algún cambio.

chelunike commented:
@oscar0310 En principio lo veo todo perfecto

oscar0310 commented:
pues si lo ves todo bien me lo puedes aprobar @chelunike y @gosema me puedes echar un vistazo cuando puedas.

chelunike reviewed (approved):
👍

gosema commented on lib/cve.pm:
```diff
@@ -0,0 +1,21 @@
+package cve;
+use Moo;
+use strictures 2;
+use namespace::clean;
```
¿El commit que has hecho se corresponde con lo que agregas de código?

gosema commented on lib/cve.pm:
```diff
@@ -0,0 +1,21 @@
+use strictures 2;
+use namespace::clean;
+
+use Types::Common::String qw(NonEmptyStr);
```
¿El titulo del commit se corresponde con el código?

gosema commented on lib/cve.pm:
```diff
@@ -0,0 +1,21 @@
+use Types::Common::String qw(NonEmptyStr);
+use Types::Standard qw(ArrayRef);
+
+has description => ( #CVE description
```
¿Por que has tomado la decisón de que sean strings no vacios y requeridos?

gosema reviewed (changes requested):
Buenas tardes @oscar0310, siento mucho no contestar antes a la revisión, se me había pasado entre tantas notificaciones.
Veo que has seguido una buena iteración de creación de issues según DDD y además el código que proporcionas satisface las necesidades de la historia de usuario.
Sin embargo, no veo mucha coherencia entre algunos commits que has hecho para agregar partes de código y te recomendaría que los otros commits que te he comentado asociado a las partes de código justifiques porque has tomado esas decisiones más que explicar que es lo que haces.

oscar0310 committed:
#13 eliminando cve.pm

Se elimina el archivo cve.pm para justificar correctamente el código en los commits

oscar0310 committed:
#13 creando la entidad cve

usando Moo como se acordó en el issue #7 de cara al modelo a la orientación de objetos, usando stricture 2 para facilitar la detección de errores y namespace::clean que evita colisiones de nombres y que funciones internas se usen como métodos públicos

oscar0310 committed:
#13 creado el atributo description de la entidad CVE

contendrá la descripción del CVE, se establece que sea un NonEmptyStr porque no tiene sentido una descripción vacía, por esto se añade el tipo NonEmptyStr y es requerido porque un CVE sin una descripción de lo que hace no sirve.

oscar0310 committed:
#13 creado el atributo links de la entidad CVE

contendrá la links  del CVE para explotar la vulnerabilidad,  se establece que sea un Array de NonEmptyStr porque es un conjunto de links los cuales no pueden ser vacios, por esto se añade el tipo ArrayRef  y es requerido porque un CVE sin los links para explotar la vuylnerabilidad no sirve.

oscar0310 committed:
#13 resolviendo los comentarios de gosema

 Se ha creado una nueva secuencia de commits siguiendo las recomendaciones de gosema justificando el código

oscar0310 commented:
Buenas @gosema y @chelunike , he actualizado la secuencia de commits para justificar el código, me podríais decir que os parece cuando podáis.

gosema commented:
> Buenas @gosema y @chelunike , he actualizado la secuencia de commits para justificar el código, me podríais decir que os parece cuando podáis.

Veo mucho mejor ahora el código, lo haces más legible para el desarrollado y justificas las decisiones que has tomado. Tienes mi visto bueno.

chelunike reviewed (approved):
:+1: Todo lo veo bien :)

JJ reviewed (approved):
👍
